blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpAggregatedMultiIngestionHandler.java

```diff
@@ -0,0 +1,103 @@
+package com.rackspacecloud.blueflood.inputs.handlers;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Timer;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.gson.*;
+import com.netflix.astyanax.connectionpool.exceptions.ConnectionException;
+import com.rackspacecloud.blueflood.http.DefaultHandler;
+import com.rackspacecloud.blueflood.http.HttpRequestHandler;
+import com.rackspacecloud.blueflood.inputs.handlers.wrappers.Bundle;
+import com.rackspacecloud.blueflood.io.Constants;
+import com.rackspacecloud.blueflood.tracker.Tracker;
+import com.rackspacecloud.blueflood.types.MetricsCollection;
+import com.rackspacecloud.blueflood.utils.Metrics;
+import com.rackspacecloud.blueflood.utils.TimeValue;
+import org.jboss.netty.channel.ChannelHandlerContext;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+
+public class HttpAggregatedMultiIngestionHandler implements HttpRequestHandler {
+
+    private static final Logger log = LoggerFactory.getLogger(HttpAggregatedMultiIngestionHandler.class);
+
+    private static final Timer handlerTimer = Metrics.timer(HttpAggregatedMultiIngestionHandler.class, "HTTP aggregated multi metrics ingestion timer");
+    private static final Counter requestCount = Metrics.counter(HttpAggregatedMultiIngestionHandler.class, "HTTP aggregated multi Request Count");
+
+    private final HttpMetricsIngestionServer.Processor processor;
+    private final TimeValue timeout;
+
+    public HttpAggregatedMultiIngestionHandler(HttpMetricsIngestionServer.Processor processor, TimeValue timeout) {
+        this.processor = processor;
+        this.timeout = timeout;
+    }
+
+    // our own stuff.
+    @Override
+    public void handle(ChannelHandlerContext ctx, HttpRequest request) {
+
+        Tracker.track(request);
+
+        final Timer.Context timerContext = handlerTimer.time();
+
+        // this is all JSON.
+        final String body = request.getContent().toString(Constants.DEFAULT_CHARSET);
+        try {
+            // block until things get ingested.
+            requestCount.inc();
+            List<Bundle> bundleList = createBundleList(body);
+            for (Bundle bundle : bundleList){
+                MetricsCollection collection = new MetricsCollection();
+                collection.add(PreaggregateConversions.buildMetricsCollection(bundle));
+                ListenableFuture<List<Boolean>> futures = processor.apply(collection);
+                List<Boolean> persisteds = futures.get(timeout.getValue(), timeout.getUnit());
```

Comment:
This looks like you are going to block for each bundle in the bundleList.  Isn't that going to be very slow?  It would probably be create a list of metrics, instead of a list of bundles and then simultaneously block on each individual metric, rather than each individual bundle.